### PR TITLE
Move auto sync to overview page

### LINKED ISSUE #771: Move "auto sync" setting from settings page to overview page
Make auto-sync more obvious for users by moving the option to the overview default tab

Proposed mock-up of Overview tab:
Screen Shot 2022-01-21 at 1 00 17 PM
Remove from Settings page:
Screen Shot 2022-01-21 at 9 58 50 AM

### LINKED ISSUE #778: No notification when spending password is changed
There shoudl be a notification of the format "Spending password updated successfully" when spending password is changed.

### PR DESCRIPTION
**Pull request by Sirmorrison**
Fix #771 
Fix #778

This Pr move the auto sync functionality from the settings page to the overview page.
image

### CODE REVIEW

File: ui/page/overview/overview_layout.go [hunk 1]
```diff
@@ -246,6 +247,12 @@ func (pg *AppOverviewPage) HandleUserInteractions() {
 		pg.isBackupModalOpened = true
 	}
 
+	autoSync := pg.WL.Wallet.ReadBoolConfigValueForKey(load.AutoSyncConfigKey)
+	pg.autoSync.SetChecked(false)
```
Comment (itswisdomagain) [inline ui/page/overview/overview_layout.go]: `pg.autoSync` should be `pg.autoSyncSwitch` so it is immediately obvious what that property is. `autoSync` as used in line 250 makes sense to represent a boolean variable, not a switch.

File: ui/page/overview/overview_layout.go [hunk 2]
```diff
@@ -246,6 +247,12 @@ func (pg *AppOverviewPage) HandleUserInteractions() {
...
+	if autoSync {
+		pg.autoSync.SetChecked(autoSync)
+	}
```
Comment (itswisdomagain) [inline ui/page/overview/overview_layout.go]: Don't need the if statement. Can just always `SetChecked(autoSync)`.

File: ui/page/overview/wallet_status.go [hunk 3]
```diff
@@ -179,7 +179,14 @@ func (pg *AppOverviewPage) syncDormantContent(gtx layout.Context, uniform layout
...
 				if pg.WL.MultiWallet.IsSynced() {
-					return pg.connectionPeer(gtx)
+					return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return pg.connectionPeer(gtx)
+						}),
+						layout.Rigid(func(gtx C) D {
+							return pg.autoSyncSec(gtx)
+						}),
+					)
```
Comment (itswisdomagain) [inline ui/page/overview/wallet_status.go]: Do we only want to show the autoSync switch _after_ sync completes?

File: ui/page/overview/sync_details.go [hunk 4]
```diff
@@ -50,6 +53,20 @@ func (pg *AppOverviewPage) connectionPeer(gtx layout.Context) layout.Dimensions
 	)
 }
 
+func (pg *AppOverviewPage) autoSyncSec(gtx layout.Context) layout.Dimensions {
```
Comment (itswisdomagain) [inline ui/page/overview/sync_details.go]: `layoutAutoSyncSection`

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: ui/page/overview/wallet_status.go (hunk 3 above)
Comment (Sirmorrison) [reply ui/page/overview/wallet_status.go]: Yes from the mock up.

**Review by itswisdomagain — APPROVED** [review]
Looks good. Just one area to clarify before merging, relating to when the auto sync switch should be made to display on the overview UI.

I think it's poor UX to make the user wait till sync completes (which can take some time for the initial blocks download) before they can disable or enable auto-sync. I feel they should be always able to toggle that setting at anytime, or at least, when sync isn't active such as when it is canceled/stopped not just when it completes.

File: ui/page/settings_page.go [hunk 5]
```diff
@@ -92,11 +90,6 @@ func NewSettingsPage(l *load.Load) *SettingsPage {
 
 	pg.backButton, pg.infoButton = components.SubpageHeaderButtons(l)
 
-	// pg.peerLabel = l.Theme.Body1("")
-	// pg.peerLabel.Color = l.Theme.Color.GrayText2
-
-	// pg.agentLabel = l.Theme.Body1("")
-	// pg.agentLabel.Color = l.Theme.Color.GrayText2
```
Comment (dreacot) [inline ui/page/settings_page.go:99]: remove comments

**Review by dreacot — COMMENTED** [review]
based on wisom's comments. looks like it's yet to be implemented https://github.com/planetdecred/godcr/pull/789#pullrequestreview-888544016

**Comment by Sirmorrison** [comment]
> based on wisom's comments. looks like it's yet to be implemented [#789 (review)](https://github.com/planetdecred/godcr/pull/789#pullrequestreview-888544016)

Comment will need a ui update. The design is currently based on the provided mockup by @monsa00

File: ui/page/settings_page.go (hunk 5 above)
Comment (Sirmorrison) [reply ui/page/settings_page.go:99]: There are no commented codes... these was actually removed not added.

Comment (dreacot) [reply ui/page/settings_page.go:99]: Oh, my bad

**Review by dreacot — APPROVED** [review]

**Pull request merged**